storage/src/mmr/bitmap.rs

```diff
@@ -0,0 +1,315 @@
+        Self::new()
+    }
+}
+
+impl<H: CHasher> Bitmap<H> {
+    const CHUNK_SIZE: usize = H::Digest::SERIALIZED_LEN;
+
+    /// Return a new empty bitmap.
+    pub fn new() -> Self {
+        Bitmap {
+            bitmap: vec![0u8; H::Digest::SERIALIZED_LEN],
+            next_bit: 0,
+            mmr: Mmr::new(),
+        }
+    }
+
+    /// Return the number of bits currently stored in the bitmap.
+    pub fn bit_count(&self) -> u64 {
+        (self.bitmap.len() * 8 - Self::CHUNK_SIZE * 8 + self.next_bit) as u64
+    }
+
+    /// Return the last chunk of the bitmap as a slice.
+    fn last_chunk(&self) -> &[u8] {
+        let len = self.bitmap.len();
+        &self.bitmap[len - Self::CHUNK_SIZE..len]
+    }
+
+    /// Return the last chunk of the bitmap as a mutable slice.
+    fn last_chunk_mut(&mut self) -> &mut [u8] {
+        let len = self.bitmap.len();
+        &mut self.bitmap[len - Self::CHUNK_SIZE..len]
+    }
+
+    /// Commit the last chunk of the bitmap to the Merkle tree and initialize the next chunk.
+    fn commit_last_chunk(&mut self, hasher: &mut H) {
+        let chunk = H::Digest::try_from(self.last_chunk()).unwrap();
+        self.mmr.add(hasher, &chunk);
+        self.next_bit = 0;
+        self.bitmap.extend(vec![0u8; Self::CHUNK_SIZE]);
+    }
+
+    /// Efficiently add a digest-sized chunk of bits to the bitmap. Assumes we are at a chunk
+    /// boundary (that is, `self.next_bit` is 0) and panics otherwise.
+    pub fn append_chunk_unchecked(&mut self, hasher: &mut H, chunk: &H::Digest) {
+        assert!(
+            self.next_bit == 0,
+            "cannot add chunk when not chunk aligned"
+        );
+
+        self.last_chunk_mut().copy_from_slice(chunk.as_ref());
+        self.commit_last_chunk(hasher);
+    }
+
+    /// Efficiently add a byte's worth of bits to the bitmap. Assumes self.next_bit is currently
+    /// byte aligned, and panics otherwise.
+    pub fn append_byte_unchecked(&mut self, hasher: &mut H, byte: u8) {
+        assert!(
+            self.next_bit % 8 == 0,
+            "cannot add byte when not byte aligned"
+        );
+
+        let chunk_byte = self.next_bit / 8;
+        self.last_chunk_mut()[chunk_byte] = byte;
+        self.next_bit += 8;
+        if self.next_bit == Self::CHUNK_SIZE * 8 {
+            self.commit_last_chunk(hasher);
+        }
+    }
+
+    /// Add a single bit to the bitmap.
+    pub fn append(&mut self, hasher: &mut H, bit: bool) {
+        if bit {
+            let chunk_byte = self.next_bit / 8;
+            self.last_chunk_mut()[chunk_byte] |= self.mask();
+        }
+        self.next_bit += 1;
+        if self.next_bit == Self::CHUNK_SIZE * 8 {
+            self.commit_last_chunk(hasher);
+        }
+    }
+
+    /// Convert a bit offset into a bit mask.
+    fn mask_for(bit_offset: u64) -> u8 {
+        1 << (bit_offset % 8)
+    }
+
+    /// Convert the next_bit offset into a bit mask.
+    fn mask(&self) -> u8 {
+        1 << (self.next_bit % 8)
+    }
```

Comment:
nit: I think since this function is not super obvious what it does from the name and is only used once, delete the function and prefer using `Self::mask_for(self.next_bit)` instead

Reply:
Removed function and changed to use `Self::mask_for(self.next_bit as u64)`